workflow/controller/operator.go

```diff
@@ -976,9 +976,10 @@ func (woc *wfOperationCtx) podReconciliation(ctx context.Context) error {
 			// If the node is pending and the pod does not exist, it could be the case that we want to try to submit it
 			// again instead of marking it as an error. Check if that's the case.
 			// Node will be in pending state without Pod create if Node is waiting for Synchronize lock
-			if node.Pending() && !node.IsPodCreated() {
+			if node.Pending() && node.GetPendingReason() == wfv1.WaitingForSyncLock {
```

Comment:
should this be `||` not `&&`?

Reply:
should be &&

Reply:
The node should not fail/error if the node is in a pending state with WaitforSyncLock.

Reply:
are you sure? `&&` means that any node that is pending and not waiting for lock may be marked as error

Reply:
yes. I am sure.  There is another scenario (pod. is not updated to informer) which will cover in `recentlyStarted`

Reply:
@simster7 - can I request another opinion on this? I feel like I'm going mad.